### ISSUE TITLE
AccessDeniedException in device instantiation unless access to **all** Amazon Braket regions is allowed 

### ISSUE THREAD

**Issue by sterseba** [qiskit-community/qiskit-braket-provider#114]
The provider only allows to create an instance of an Amazon Braket backend if the calling AWS identity has permissions to access **every** Amazon Braket region. This is not intuitive, not needed from a Braket API perspective and a problem for users who require access to a specific device but - for whatever reason - only have permissions to access the AWS region of that device.

### Steps to reproduce the problem
Assume you have permissions to use Braket in every region except us-west-1. This may be due to a service control policy at the account level or an IAM policy attached to your identity. For simplicity, attach the following as inline policy to the IAM identity used to access Braket:
```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "DenyBraketActionsInUsWest1",
      "Effect": "Deny",
      "Action": [
        "braket:*"
      ],
      "Resource": [
        "arn:aws:braket:us-west-1:*:*/*"
      ]
    }
  ]
}
```

Then, execute the following script:

```python
from qiskit_braket_provider import AWSBraketProvider

provider = AWSBraketProvider()

device_types = ["SIMULATOR", "QPU"]
backends = ["SV1", "Lucy", "Harmony", "Aspen-M-3"]

for device_type in device_types:
    try:
        result = provider.backends(statuses=["ONLINE"], types=[device_type])
        print(f"Search for device type {device_type} successful", "\n")
    except Exception as e:
        print(f"Search for device type {device_type} failed due to {type(e)}", "\n")

for backend in backends:
    try:
        result = provider.get_backend(backend)
        print(f"Successfully instantiated {backend}", "\n")
    except Exception as e:
        print(f"Failed instantiating {backend} due to {type(e)}", "\n")
```

### What is the current behavior?

Above script outputs:
```
Search for device type SIMULATOR successful 

Search for device type QPU failed due to <class 'botocore.errorfactory.AccessDeniedException'> 

Failed instantiating SV1 due to <class 'botocore.errorfactory.AccessDeniedException'> 

Failed instantiating Lucy due to <class 'botocore.errorfactory.AccessDeniedException'> 

Failed instantiating Harmony due to <class 'botocore.errorfactory.AccessDeniedException'> 

Failed instantiating Aspen-M-3 due to <class 'botocore.errorfactory.AccessDeniedException'> 
```

- As expected, access to the Aspen-M-3 device (`provider.get_backend("Aspen-M-3")`) is denied because it is located in us-west-1.
- As also expected, the search for simulator devices (`provider.backends(statuses=["ONLINE"], types=["SIMULATOR"])`)  is successful because these devices exist in every Braket region.
- The search for QPUs (`provider.backends(statuses=["ONLINE"], types=["QPU"])`) fails which is _okay_ because access to one of the QPUs is denied.
- **However, none of the devices which are in regions other than us-west-1 can be instantiated either.** Not even a simulator although the search for simulators is successful. This is because in `get_backend` the provider calls the `braket:SearchDevices` API  which fails with:
```
An error occurred (AccessDeniedException) when calling the SearchDevices operation: User: arn:aws:sts::XXXXXXXXXXXX:assumed-role/AmazonBraketServiceSageMakerNotebookRole-XXXXXX/SageMaker is not authorized to perform: braket:SearchDevices on resource: arn:aws:braket:us-west-1:XXXXXXXXXXXX:/devices with an explicit deny
```

### What is the expected behavior?

The same behavior as when using the Braket SDK is expected.

The following code only fails in the attempt to instantiate the Aspen-M-3 or to search for all QPUs online.
All other devices can be instantiated successfully.

```python
from braket.aws import AwsDevice

device_types = ["SIMULATOR", "QPU"]
backends = [
    "arn:aws:braket:::device/quantum-simulator/amazon/sv1",
    "arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy",
    "arn:aws:braket:us-east-1::device/qpu/ionq/Harmony",
    "arn:aws:braket:us-west-1::device/qpu/rigetti/Aspen-M-3"
]

for device_type in device_types:
    try:
        result = AwsDevice.get_devices(statuses=['ONLINE'], types=[device_type])
        print(f"Search for device type {device_type} successful", "\n")
    except Exception as e:
        print(f"Search for device type {device_type} failed due to {type(e)}", "\n")

for backend in backends:
    try:
        result = AwsDevice(backend)
        print(f"Successfully instantiated {result.name}", "\n")
    except Exception as e:
        print(f"Failed instantiating {backend} due to {type(e)}", "\n")
```

```
Search for device type SIMULATOR successful 

Search for device type QPU failed due to <class 'botocore.errorfactory.AccessDeniedException'> 

Successfully instantiated SV1 

Successfully instantiated Lucy 

Successfully instantiated Harmony 

Failed instantiating arn:aws:braket:us-west-1::device/qpu/rigetti/Aspen-M-3 due to <class 'botocore.errorfactory.AccessDeniedException'> 
```

**Comment by kshitijc** [comment]
Thanks for reporting this issue @sterseba and for including all the details! They are super helpful.

The reason why we are able to instantiate the device in the SDK is because we are supplying the device arn (which has a region specified), allowing the SDK to target the specific region. In case of the Qiskit Braket Provider, we only supply the device name and not the arn. Hence we need to look across the devices to match the name. 
I think even for the search for device type QPU in the SDK, it might make more sense for it to return a filtered down list of devices in all accessible regions instead of throwing an exception. 

We could achieve this by handling the AccessDenied exception in the AwsDevice class [here](https://github.com/aws/amazon-braket-sdk-python/blob/6f79c45e396141567d88cc8c9bc6c82b453bd2d1/src/braket/aws/aws_device.py#L563-L579). Additionally, we should also log the AccessDeniedException, to provide customers some visibility into these errors. We could employ loggers in a similar manner to how they are employed in the AwsQuantumTask class [here](https://github.com/aws/amazon-braket-sdk-python/blob/6f79c45e396141567d88cc8c9bc6c82b453bd2d1/src/braket/aws/aws_quantum_task.py#L200). 

This should fix the issue here, and also allow SDK users to be able to search for QPUs in a subset of regions. 

Code contributions are highly encouraged! We would love to have you, or anyone else who might be interested in tackling this issue, help fix this issue!

**Comment by kshitijc** [comment]
Until the fix described above is implemented, it is possible to work around this issue by instantiating the `AWSBraketBackend` object directly instead of querying the `AwsBraketProvider`. 

So instead of doing 

```
from qiskit_braket_provider import AWSBraketProvider

provider = AWSBraketProvider()
device = provider.get_backend("Aspen-M-3")
```

you could instead do

```
from braket.aws import AwsDevice
from qiskit_braket_provider import AWSBraketBackend

device = AWSBraketBackend(AwsDevice("arn:aws:braket:us-west-1::device/qpu/rigetti/Aspen-M-3"))
```

**Comment by kshitijc** [comment]
This has been fixed in the SDK release [v1.56.1](https://github.com/amazon-braket/amazon-braket-sdk-python/releases/tag/v1.56.1)
